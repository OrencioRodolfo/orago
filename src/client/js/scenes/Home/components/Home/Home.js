/* global fetch */
import React, { Component } from 'react';
import styled from 'styled-components';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { getAbout } from '../../../../actions';
import RestrictedContainer from '../../../../components/Containers/Restricited/Restricited';
import Header from '../Header/Header';
import Subhead from '../Subhead/Subhead';
import Footer from '../Footer/Footer';

const HomeContainer = styled.div`
  font-family: ${props => props.theme.fontFamily('primary')};
  display: flex;
  flex-direction: column;
  height: 100vh;
  
  > .main-container {
    flex-grow: 1;
    display: flex;
    align-items: center;
  }
`;


class Home extends Component {
  componentDidMount() {
    fetch(`${API_URL}/about`)
      .then(response => response.json())
      .then((response) => {
        this.props.getAbout(response);
      });
  }

  render() {
    return (
      <HomeContainer>
        <Header />
        <Subhead />
        <RestrictedContainer className="main-container">Rodolfo Gon√ßalves</RestrictedContainer>
        <Footer />
      </HomeContainer>
    );
  }
}

Home.propTypes = {
  getAbout: PropTypes.func.isRequired,
};

function matchDispatchToProps(dispatch) {
  return bindActionCreators({
    getAbout,
  }, dispatch);
}

export default connect(null, matchDispatchToProps)(Home);

export {
  Home,
};
