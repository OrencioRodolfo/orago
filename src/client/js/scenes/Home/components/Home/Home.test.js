import React from 'react';
import { shallow } from 'enzyme';
import Home from './Home';

describe('App component', () => {
  it('Must render properly', () => {
    const container = shallow(
      <Home />,
    );

    expect(container).toMatchSnapshot();
  });
});
