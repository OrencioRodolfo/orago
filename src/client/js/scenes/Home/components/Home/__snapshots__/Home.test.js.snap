// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component Must render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home
    description=""
    getAbout={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Header />,
        <Subhead />,
        <Container
          className="grow"
        >
          <styled.p>
            
          </styled.p>
        </Container>,
        <Footer />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "children": <styled.p>
            
          </styled.p>,
          "className": "grow",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Header />,
          <Subhead />,
          <Container
            className="grow"
          >
            <styled.p>
              
            </styled.p>
          </Container>,
          <Footer />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "children": <styled.p>
              
            </styled.p>,
            "className": "grow",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": "",
            },
            "ref": null,
            "rendered": "",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
